src/OutputWriters/netcdf_output_writer.jl

```diff
@@ -150,7 +150,7 @@ function NetCDFOutputWriter(model, outputs; interval=nothing, frequency=nothing,
 
     # Initiates the output file with dimensions
     write_grid_and_attributes(model; filename=filename, compression=compression,
-                              attributes=global_attributes, slice_kw...)
+                              attributes=global_attributes, mode=mode, slice_kw...)
```

Comment:
@suyashbire1 Hmmm is this what we want?

If the file does not already exist, then yes you should write grid and attributes. But if the file already exists then maybe we shouldn't be writing grid and attributes?

So maybe using `mode="c"` made sense here so it always clobbers when writing grid and attributes. But then `NetCDFOutputWriter` always clobbers right?

Reply:
> So maybe using mode="c" made sense here so it always clobbers when writing grid and attributes. But then NetCDFOutputWriter always clobbers right?

Yes, `NetCDFOutputWriter`'s default behaviour is to clobber.

> If the file does not already exist, then yes you should write grid and attributes. But if the file already exists then maybe we shouldn't be writing grid and attributes?

While this'll be a good feature to add, I don't think that's the issue here. The error looks like a permission denied issue. If I am not wrong, the CPU tests run first followed by the GPU tests. So the file is created by the CPU test and the GPU test is trying to clobber it but can't because of lack of permissions.

> Not sure if has something to do with clobber vs. append but maybe giving different filenames for CPU and GPU will help? I.e. nc_filename = "dump_test_$(typeof(arch)).nc"

Shouldn't be an append vs clobber issue because default behaviour is clobber. I think GPU tests don't have permissions to clobber files created by CPU tests. Does that make sense? Anyway, I'll send commits today to check if arch-specific names help.